src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -650,45 +652,57 @@ def __repr__(self) -> str:
         return f"<OSD>(osd_id={self.osd_id}, draining={self.draining})"
 
 
-class OSDQueue(Set):
+class OSDQueue(Queue):
 
     def __init__(self):
         super().__init__()
+        self.lock = threading.Lock()
```

Comment:
It's a pity that Python doesn't come with a standard RWLock, as here we have clearly defined read (`as_*`) and write (`put`, `rm`) locks...

Reply:
yep